packages/shopify-app-express/src/auth/index.ts

```diff
@@ -1,30 +1,28 @@
-import {Request, Response} from 'express';
+import {NextFunction, Request, RequestHandler, Response} from 'express';
 
-import {AttachAuthParams} from './types';
-import {authBegin} from './auth-begin';
-import {authCallback} from './auth-callback';
-
-export function attachAuth({
-  api,
-  config,
-  subApp,
-  afterAuth,
-}: AttachAuthParams): void {
-  subApp.get(config.auth.path, async (req: Request, res: Response) => {
-    await config.logger.debug('Initiating auth begin request');
+import {ApiAndConfigParams} from '../types';
+import {redirectToAuth} from '../redirect-to-auth';
 
-    return authBegin({req, res, api, config});
-  });
+import {authCallback} from './auth-callback';
+import {AuthMiddleware} from './types';
 
-  subApp.get(config.auth.callbackPath, async (req: Request, res: Response) => {
-    await config.logger.debug('Initiating auth callback request');
+export function auth({api, config}: ApiAndConfigParams): AuthMiddleware {
+  return {
+    begin(): RequestHandler {
+      return async (req: Request, res: Response) =>
+        redirectToAuth({req, res, api, config});
+    },
+    callback(): RequestHandler {
```

Comment:
since there are no arguments here, is there any reason this can't be the actual express middleware function itself?

Reply:
I was wondering about that - we've used this pattern of making all of the middlewares we expose functions to make it possible to introduce arguments to them in the future without making it a breaking change.